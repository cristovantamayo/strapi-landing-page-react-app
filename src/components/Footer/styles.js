import styled, { css } from "styled-components";
import { theme } from "../../styles/theme";

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    a {
      color: inherit;
      text-decoration: nome;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
