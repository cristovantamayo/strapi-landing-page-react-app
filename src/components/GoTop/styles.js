import styled, { css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4.8rem;
    height: 4.8rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease-in-out;

    &:hover {
      opacity: 1;
      background: red;
    }
  `}
`;
