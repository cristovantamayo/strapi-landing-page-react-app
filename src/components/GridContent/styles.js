import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    max-width: 60rem;
    margin: 0 auto;

    @media ${theme.media.lessMedium} {
      max-width: 40rem;
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xHuge};
  `}
`;
