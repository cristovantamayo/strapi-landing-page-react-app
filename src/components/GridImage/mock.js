export default {
  background: false,
  title: "My grid",
  description:
    "Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde",
  grid: [
    {
      altText: "Teste 1",
      srcImg:
        "https://images.unsplash.com/photo-1506748686214-e9df14d4d9d0?w=800&h=800",
    },
    {
      altText: "Teste 2",
      srcImg:
        "https://images.unsplash.com/photo-1496483648148-47c686dc86a8?w=800&h=800",
    },
    {
      altText: "Teste 3",
      srcImg:
        "https://images.unsplash.com/photo-1469474968028-56623f02e42e?w=800&h=800",
    },
    {
      altText: "Teste 4",
      srcImg:
        "https://images.pexels.com/photos/460621/pexels-photo-460621.jpeg?w=800&h=800",
    },
    {
      altText: "Teste 5",
      srcImg:
        "https://images.pexels.com/photos/210186/pexels-photo-210186.jpeg?w=800&h=800",
    },
    {
      altText: "Teste 6",
      srcImg:
        "https://images.unsplash.com/photo-1496483648148-47c686dc86a8?w=800&h=800",
    },
  ],
};
