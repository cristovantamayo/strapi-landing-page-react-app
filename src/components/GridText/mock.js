export default {
  background: false,
  title: "My grid",
  description: `Atque dolorum, quibusdam
    quisquam, quas, voluptate
    quos laboriosam voluptatum,
    quisquam quisquam, quas,
    voluptate quos laboriosam
    voluptatum.`,
  grid: [
    {
      title: "Teste 1",
      description: `Lorem ipsum dolor sit,
      amet consectetur adipisicing elit.
      Debitis cum delectus molestias.
      Atque doloribus nobis laudantium
      esse ut, non commodi maxime
      distinctio veritatis unde,
      reprehenderit minus ad dolores
      provident maiores.`,
    },
    {
      title: "Teste 2",
      description: `Lorem ipsum dolor sit,
      amet consectetur adipisicing elit.
      Debitis cum delectus molestias.
      Atque doloribus nobis laudantium
      esse ut, non commodi maxime
      distinctio veritatis unde,
      reprehenderit minus ad dolores
      provident maiores.`,
    },
    {
      title: "Teste 3",
      description: `Lorem ipsum dolor sit,
      amet consectetur adipisicing elit.
      Debitis cum delectus molestias.
      Atque doloribus nobis laudantium
      esse ut, non commodi maxime
      distinctio veritatis unde,
      reprehenderit minus ad dolores
      provident maiores.`,
    },
  ],
};
