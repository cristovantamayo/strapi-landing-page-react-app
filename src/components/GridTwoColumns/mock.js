export default {
  title: "Grid Two Columns",
  text: "Lorem ipsum dolor sit amet consectetur adipisicing elit",
  srcImg: "assets/images/javascript.svg",
};
