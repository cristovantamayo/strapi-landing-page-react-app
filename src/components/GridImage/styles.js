import styled, { css } from "styled-components";
import { Title as HeadingContainer } from "../Heading/styles";
import { Container as TextComponent } from "../TextComponent/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 128rem;
    margin: 0 auto;
    left: 0;
    right: 0;
    & > ${TextComponent} {
      margin-bottom: ${theme.spacings.xHuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    max-width: 120rem;
    margin: 0 auto;

    @media ${theme.media.lessMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    width: 100%;
    height: 280px;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    height: 100%;
    width: 100%;
    object-fit: cover;

    &:hover {
      transform: scale(1.2) rotate(10deg);
      transition: transform 0.5s;
    }
  `}
`;
