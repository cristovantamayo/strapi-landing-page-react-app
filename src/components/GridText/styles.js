import styled, { css } from "styled-components";
import { Title as HeadingContainer } from "../Heading/styles";
import { Container as TextComponent } from "../TextComponent/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    max-width: 120rem;
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xHuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.xHuge};
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xHuge};
      top: -3.5rem;
      left: -4rem;
      transform: rotate(10deg);
  `}
`;
