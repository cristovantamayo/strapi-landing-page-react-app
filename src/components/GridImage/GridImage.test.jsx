import { screen } from "@testing-library/react";
import { describe, expect } from "@jest/globals";
import { renderTheme } from "../../styles/render-theme";
import { GridImage } from ".";
import mock from "./mock";

describe("<GridImage />", () => {
  it("should render GridText with background", () => {
    const { container } = renderTheme(<GridImage {...mock} />);
    expect(container).toMatchSnapshot();
  });
  it("should render GridText without background", () => {
    const { container } = renderTheme(
      <GridImage {...mock} background={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
