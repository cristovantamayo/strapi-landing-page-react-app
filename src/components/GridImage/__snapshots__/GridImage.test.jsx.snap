// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage /> should render GridText with background 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.8rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  max-width: 128rem;
  margin: 0 auto;
  left: 0;
  right: 0;
}

.c2>.c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
  max-width: 120rem;
  margin: 0 auto;
}

.c7 {
  overflow: hidden;
  width: 100%;
  height: 280px;
}

.c8 {
  height: 100%;
  width: 100%;
  object-fit: cover;
}

.c8:hover {
  transform: scale(1.2) rotate(10deg);
  transition: transform 0.5s;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

@media (max-width: 768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (max-width: 768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          My grid
        </h1>
        <div
          class="c4 c5"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Teste 1"
              class="c8"
              src="https://images.unsplash.com/photo-1506748686214-e9df14d4d9d0?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 2"
              class="c8"
              src="https://images.unsplash.com/photo-1496483648148-47c686dc86a8?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 3"
              class="c8"
              src="https://images.unsplash.com/photo-1469474968028-56623f02e42e?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 4"
              class="c8"
              src="https://images.pexels.com/photos/460621/pexels-photo-460621.jpeg?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 5"
              class="c8"
              src="https://images.pexels.com/photos/210186/pexels-photo-210186.jpeg?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 6"
              class="c8"
              src="https://images.unsplash.com/photo-1496483648148-47c686dc86a8?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridImage /> should render GridText without background 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.8rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  max-width: 128rem;
  margin: 0 auto;
  left: 0;
  right: 0;
}

.c2>.c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
  max-width: 120rem;
  margin: 0 auto;
}

.c7 {
  overflow: hidden;
  width: 100%;
  height: 280px;
}

.c8 {
  height: 100%;
  width: 100%;
  object-fit: cover;
}

.c8:hover {
  transform: scale(1.2) rotate(10deg);
  transition: transform 0.5s;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

@media (max-width: 768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (max-width: 768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          My grid
        </h1>
        <div
          class="c4 c5"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Teste 1"
              class="c8"
              src="https://images.unsplash.com/photo-1506748686214-e9df14d4d9d0?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 2"
              class="c8"
              src="https://images.unsplash.com/photo-1496483648148-47c686dc86a8?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 3"
              class="c8"
              src="https://images.unsplash.com/photo-1469474968028-56623f02e42e?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 4"
              class="c8"
              src="https://images.pexels.com/photos/460621/pexels-photo-460621.jpeg?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 5"
              class="c8"
              src="https://images.pexels.com/photos/210186/pexels-photo-210186.jpeg?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 6"
              class="c8"
              src="https://images.unsplash.com/photo-1496483648148-47c686dc86a8?w=800&h=800"
            />
            <div
              class="c4 c5"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
